test/TesterInternal/MembershipTests/MembershipTableTestsBase.cs

```diff
@@ -1,72 +1,76 @@
 using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Net;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Orleans;
-using Orleans.AzureUtils;
+using Orleans.Messaging;
 using Orleans.Runtime;
 using Orleans.Runtime.Configuration;
 using UnitTests.StorageTests;
 using Xunit;
-using Assert = Microsoft.VisualStudio.TestTools.UnitTesting.Assert;
 
 namespace UnitTests.MembershipTests
 {
-    [TestClass]
-    public abstract class MembershipTableTestsBase : IClassFixture<MembershipTableTestsBase.Fixture>, IDisposable
+    public abstract class MembershipTableTestsBase : IDisposable, IClassFixture<MembershipTableTestsBase.Fixture>
     {
-        public class Fixture : IDisposable
-        {
-            public Fixture()
-            {
-                TraceLogger.Initialize(new NodeConfiguration());
+        private static readonly string hostName = Dns.GetHostName();
+        private readonly TraceLogger logger;
+        private readonly IMembershipTable membershipTable;
+        private readonly IGatewayListProvider gatewayListProvider;
+        private readonly string deploymentId;
 
-                // Set shorter init timeout for these tests
-                OrleansSiloInstanceManager.initTimeout = TimeSpan.FromSeconds(20);
-            }
+        protected const string testDatabaseName = "OrleansMembershipTest";//for relational storage
 
-            public void Dispose()
-            {
-                // Reset init timeout after tests
-                OrleansSiloInstanceManager.initTimeout = AzureTableDefaultPolicies.TableCreationTimeout;
-            }
+        public class Fixture
+        {
+            public string ConnectionString;
         }
 
-        private static readonly string hostName = Dns.GetHostName();
-        private TraceLogger logger;
-        private IMembershipTable membershipTable;
-        private string deploymentId;
-
-        public MembershipTableTestsBase()
+        protected MembershipTableTestsBase(Fixture fixture)
         {
+            TraceLogger.Initialize(new NodeConfiguration());
             logger = TraceLogger.GetLogger(GetType().Name, TraceLogger.LoggerType.Application);
             deploymentId = "test-" + Guid.NewGuid();
 
             logger.Info("DeploymentId={0}", deploymentId);
 
+            lock (fixture)
```

Comment:
well, in this case it shouldn't make a difference, but to make it readable to anyone but me, i'll fix it.